cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/app/etl/batch/ETLMapReduce.java

```diff
@@ -141,6 +188,16 @@ public void beforeSubmit(MapReduceContext context) throws Exception {
     job.setNumReduceTasks(0);
   }
 
+  private void addPropertiesToErrorDataset(String errorDatasetName, MapReduceContext context) {
+    if (errorDatasetName != null) {
+      Map<String, String> args = new HashMap<>();
+      org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(ERROR_SCHEMA.toString());
```

Comment:
shouldn't parse this each time, but just once.  Actually shouldn't even parse it, can just build the avro schema directly